cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md

```diff
@@ -1,4 +1,4 @@
-# libcudf C++ Developer Guide
+# libcudf C++ Developer Guide {#DEVELOPER_GUIDE}
 
 This document serves as a guide for contributors to libcudf C++ code. Developers should also refer
 to these additional files for further documentation of libcudf best practices.
```

Comment:
Do the links to "these additional files" work? Should this top section be rewritten for a web-based audience?

Reply:
Yes, these links work.